### ISSUE TITLE
[CLI] Analyze command does not create the XML output file and returns incorrect exit code when only compiler diagnostics are reported

### ISSUE THREAD

**Issue by PeterKaszab** [JosefPihrt/Roslynator#1055]
**Product and Version Used**: 
Roslynator.DotNet.Cli 0.5.0

**Steps to Reproduce**:
Create a new console project, and place the following code to Program.cs
```csharp
int a, b = 1, 0;
Console.WriteLine(a);
```

Then, run `roslynator analyze -o out.xml` on the previously created project.

**Actual Behavior**:
- `CS0219` is reported, but `out.xml` is not created.
- If the output also contains analyzer diagnostics, then `out.xml` created and contains both analyzer and compiler diagnostics.
- Exit code is `0`

**Expected Behavior**:
`out.xml` should be created even if it only contains compiler diagnostics.
- Exit code should be `1`

**Comment by PeterKaszab** [comment]
Note: this is really similar to #963, which is already fixed by #964.
